joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -161,10 +206,17 @@ def configure_robot(self, description):
         self.free_joints = {}
         self.joint_list = [] # for maintaining the original order of the joints
 
-        if robot.getElementsByTagName('COLLADA'):
+        # Get root tag to parse file format
+        root = robot.documentElement
+        if root.tagName == 'sdf':
+            self.init_sdf(robot)
+        elif root.tagName == 'COLLADA':
             self.init_collada(robot)
-        else:
+        elif root.tagName == 'urdf' or root.tagName == 'robot':
             self.init_urdf(robot)
+        else:
+            self.get_logger().error('Cannot parse file of type %s' % root.tagName)
+            return
```

Comment:
After thinking about it more, for maximum compatibility we should probably just leave this as `else`.  It is somewhat incorrect in that we'll try to parse *anything* else as URDF, but it also keeps us the most compatible with what is already there.

```suggestion
        else:
            self.init_urdf(robot)
```